source/extensions/filters/network/thrift_proxy/buffer_helper.cc

```diff
@@ -0,0 +1,200 @@
+#include "extensions/filters/network/thrift_proxy/buffer_helper.h"
+
+#include "common/common/byte_order.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ThriftProxy {
+
+int8_t BufferHelper::peekI8(Buffer::Instance& buffer, size_t offset) {
+  ASSERT(buffer.length() >= 1);
```

Comment:
What do you think about throwing an exception here if the buffer length is too small, than asserting? Even though that would mean an extra conditional branch in the critical path, the branch would always go in the same direction in the absence of bugs, so the CPU should be able to predict the branch with near-perfect accuracy.

Reply:
Seems reasonable.